src/openrct2/interface/Viewport.cpp

```diff
@@ -877,10 +913,29 @@ void viewport_paint(rct_viewport* viewport, rct_drawpixelinfo* dpi, int16_t left
     // this as well as the [x += 32] in the loop causes signed integer overflow -> undefined behaviour.
     int16_t rightBorder = dpi1.x + dpi1.width;
 
+    std::vector<paint_session*> columns;
+
+    bool useMulticore = gConfigGeneral.multicore;
+    if (viewport != window_get_main()->viewport)
```

Comment:
```suggestion
    if (window_get_main() && viewport != window_get_main()->viewport)
```

Reply:
Is it possible that it can get into viewport_paint without a main window?

Reply:
`benchgfx` does this.